examples/ble-gatt-server/src/main/java/no/nordicsemi/android/ble/ble_gatt_server/GattService.kt

```diff
@@ -0,0 +1,277 @@
+                        or BluetoothGattCharacteristic.PROPERTY_NOTIFY,
+                BluetoothGattCharacteristic.PERMISSION_READ_ENCRYPTED_MITM,  // permissions
+                description("A characteristic to be read", false) // descriptors
+        )
+        private val myGattService = service(MyServiceProfile.MY_SERVICE_UUID, myGattCharacteristic)
+
+        private val serverConnections = mutableMapOf<String, ServerConnection>()
+
+        private var currentValue: String? = null
+
+        fun setValue(value: String) {
+            currentValue = value
+            serverConnections.values.forEach { client ->
+                client.setValue(value)
+            }
+        }
+
+        override fun log(priority: Int, message: String) {
+            if (BuildConfig.DEBUG || priority == Log.ERROR) {
+                Log.println(priority, TAG, message)
+            }
+        }
+
+        override fun initializeServer(): List<BluetoothGattService> {
+            setServerObserver(this)
+
+            // In this example there's only one service, so singleton list is created.
+            return Collections.singletonList(myGattService)
+        }
+
+        override fun onServerReady() {
+            log(Log.INFO, "Gatt server ready")
+        }
+
+        override fun onDeviceConnectedToServer(device: BluetoothDevice) {
+            log(Log.DEBUG, "Device connected ${device.address}")
+
+            val clientConnection = ServerConnection()
+            clientConnection.useServer(this@ServerManager)
+
+            serverConnections[device.address] = clientConnection
+
+            if (currentValue != null) {
+                clientConnection.setValue(currentValue!!)
+            }
+        }
+
+        override fun onDeviceDisconnectedFromServer(device: BluetoothDevice) {
+            log(Log.DEBUG, "Device disconnected ${device.address}")
+            val serverConnection = serverConnections[device.address]
+            if (serverConnection != null) {
+                serverConnection.close()
+                serverConnections.remove(device.address)
+            }
+        }
+
+        /*
+         * Manages the state of an individual server connection (there can be many of these)
+         */
+        inner class ServerConnection : BleManager(context) {
+
+            private var gattCallback: GattCallback? = null
+
+            fun setValue(value: String) {
+                gattCallback?.setValue(value)
+            }
+
+            override fun getGattCallback(): BleManagerGattCallback {
+                gattCallback = GattCallback(myGattCharacteristic)
+                return gattCallback!!
+            }
+
+            private inner class GattCallback(val myGattCharacteristic: BluetoothGattCharacteristic) : BleManagerGattCallback() {
+
+                fun setValue(value: String) {
+                    setCharacteristicValue(myGattCharacteristic, value.toByteArray()).enqueue()
+                }
```

Comment:
We definitely get here for when a characteristic is set. However, I'm still not able to observe it being read.